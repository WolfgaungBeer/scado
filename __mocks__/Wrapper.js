import React from 'react';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import { Theme, defaultTheme } from '../scado';

const mockStore = configureStore();
const store = mockStore({
    scado: {
        theme: defaultTheme,
        toasts: [
            { id: '1', type: 'success', content: 'Text', onClose: () => {} },
            { id: '2', type: 'error', content: 'Text', onClose: () => {} },
        ],
    },
});


const Wrapper = ({ children }) => ( // eslint-disable-line
    <Provider store={store}>
        <Theme>{children}</Theme>
    </Provider>
);

export default Wrapper;
