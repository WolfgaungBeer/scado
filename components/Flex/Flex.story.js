import React from 'react';
import { storiesOf } from '@storybook/react';
import styled from 'styled-components';
import Theme from '../Theme/Theme';
import defaultTheme from '../Theme/defaultTheme';
import Icon from '../Icon';
import Flex from './';

import '../../node_modules/font-awesome/css/font-awesome.css';

const FlexParent = Flex.extend`
    width: 300px;
    height: 100px;
    background-color: #777777;
`;

storiesOf('Flex', module)
    .add('default', () => (
        <Theme theme={defaultTheme}>
            <FlexParent inline alignItems="center">
                <Flex flex="1 1 auto"><Icon icon="stars"/></Flex>
                <Flex flex="1 1 auto"><Icon icon="stars"/></Flex>
                <Flex flex="1 1 auto"><Icon icon="stars"/></Flex>
                <Flex flex="1 1 auto"><Icon icon="stars"/></Flex>
                <Flex flex="1 1 auto"><Icon icon="stars"/></Flex>
            </FlexParent>
        </Theme>
    ))
    .add('centered child', () => (
        <Theme theme={defaultTheme}>
            <FlexParent justifyContent="center" alignItems="center" inline>
                <Icon icon="stars" color="success" scale="l" />
            </FlexParent>
        </Theme>
    ));
