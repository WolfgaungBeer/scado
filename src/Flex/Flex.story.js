import React from 'react';
import { storiesOf } from '@storybook/react';
import { text, boolean } from '@storybook/addon-knobs';
import Flex from './';


const FlexParent = Flex.extend`
    width: 100%;
    height: 100px;
    background-color: #999999;
`;

const FlexChild = Flex.extend`
    width: 20px;
    height: 80px;
    background-color: ${props => props.theme.color.success};
`;


storiesOf('Flex', module)
    .add('default', () => (
        <FlexParent
            inline={boolean('inline', false)}
            justifyContent={text('justifyContent', 'space-around')}
            alignItems={text('alignItems', 'center')}
        >
            <FlexChild flex="0 1 auto" />
            <FlexChild flex="0 1 auto" />
            <FlexChild flex="0 1 auto" />
            <FlexChild flex="0 1 auto" />
            <FlexChild flex="0 1 auto" />
        </FlexParent>
    ))
    .add('children', () => {
        const flexChildText = text('flex', '0 1 auto');
        return (
            <FlexParent justifyContent="space-around" alignItems="center">
                <FlexChild flex={flexChildText} />
                <FlexChild flex={flexChildText} />
                <FlexChild flex={flexChildText} />
                <FlexChild flex={flexChildText} />
                <FlexChild flex={flexChildText} />
            </FlexParent>
        );
    });
