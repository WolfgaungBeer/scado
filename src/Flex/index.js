import styled from 'styled-components';
import { bool, string } from 'prop-types';

const propTypes = {
    inline: bool,
    flex: string,
    justifyContent: string,
    alignItems: string,
    className: string,
};

const defaultProps = {
    inline: undefined,
    flex: undefined,
    justifyContent: undefined,
    alignItems: undefined,
    className: undefined,
};

const Flex = styled.div`
    display: ${props => (props.inline ? 'inline-flex' : 'flex')};
    flex: ${props => props.flex || '0 1 auto'};
    justify-content: ${props => props.justifyContent};
    align-items: ${props => props.alignItems};
`;

Flex.propTypes = propTypes;
Flex.defaultProps = defaultProps;
Flex.displayName = 'Flex';

export default Flex;
