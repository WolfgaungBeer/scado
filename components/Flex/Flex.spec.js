import React from 'react';
import renderer from 'react-test-renderer';
import Wrapper from '../../__mocks__/Wrapper';
import Flex from './';

it('renders the Flex Component correctly', () => {
    const tree1 = renderer.create(
        <Wrapper>
            <Flex />
        </Wrapper>,
    ).toJSON();
    expect(tree1).toMatchSnapshot();

    const tree2 = renderer.create(
        <Wrapper>
            <Flex inline justifyContent="center" alignItems="center" />
        </Wrapper>,
    ).toJSON();
    expect(tree2).toMatchSnapshot();

    const tree3 = renderer.create(
        <Wrapper>
            <Flex flex="1 1 20%" />
        </Wrapper>,
    ).toJSON();
    expect(tree3).toMatchSnapshot();
});
