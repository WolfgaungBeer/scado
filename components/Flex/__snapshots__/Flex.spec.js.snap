// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the Flex Component correctly 1`] = `
<div
  className="sc-bdVaJa esgqVQ"
/>
`;

exports[`renders the Flex Component correctly 2`] = `
<div
  className="sc-bdVaJa ceAXfm"
/>
`;

exports[`renders the Flex Component correctly 3`] = `
<div
  className="sc-bdVaJa hNHSrk"
/>
`;
